Components fixes (#255)
intel/openfl: docs/source/openfl/components.rst

    - The *Aggregator* which receives model updates from *Collaborators* and combines them to form the global model.
    - The *Collaborator* which uses local dataset to train a global model.

The *Aggregator* is framework-agnostic, as it operates tensors in OpenFL inner representation, 
while the *Collaborator* can use deep learning frameworks as computational backend, such as `TensorFlow* <https://www.tensorflow.org/>`_ or `PyTorch* <https://pytorch.org/>`_.

Aggregator
----------

The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. 

> It orchestrates Collaborators according to the FL plan, performs model aggregation at the end of each round, 

Reviewer: ... each round and acts as a parameter server for collaborators.

> and acts as parametr server for collaborators. 

Author: ```suggestion
and acts as a parameter server for collaborators. 
```

> Model weights aggregation logic may be customized via :ref:`plugin mechanism <overriding_agg_fn>`.

Reviewer: ```suggestion
Model weight aggregation logic may be customized via :ref:`plugin mechanism <overriding_agg_fn>`.
```

    - converting deep learning framework-specific tensor objects to |productName| inner representation, and
    - exchanging model parameters with the Aggregator.

The Collaborator is created by the :ref:`Envoy <openfl_ll_components_envoy>` when a new experiment is submitted 
in the :ref:`Director-based workflow <director_workflow>`. The Collaborator should be started from CLI if a user follows the 
:ref:`Aggregator-based workflow <running_the_federation_aggregator_based>`

Every Collaborator is a unique service. The data loader is loaded with a local *shard descriptor* to perform tasks 

> included in an FL experiment. At the end of the training task, weights are extracted and sent to the central node 

Author: ```suggestion
included in an FL experiment. At the end of the training task, weight tensors are extracted and sent to the central node 
```

and aggregated.

Converting tensor objects is handled by :ref:`framework adapter <framework_adapter>` plugins. 
Included in |productName| are framework adapters for PyTorch and TensorFlow 2.x. 

> The list of framework adapters is extensible. User can contribute new framework adapters for deep learning frameworks 

Reviewer: `User can contribute ... they would like see ...` or  `You can contribute ... you would like see ...`

Author: fixed thanks

you would like see supported in |productName|. 

.. _openfl_ll_components:

Long-Lived Components
======================

These components were introduced to support the :ref:`Director-based workflow <director_workflow>`. 
	
    - The *Director* is the central node of the federation. This component starts an *Aggregator* for each experiment, broadcasts experiment archive to connected collaborator nodes, and provides updates on the status.
    - The *Envoy* runs on collaborator nodes and is always connected to the *Director*. When the *Director* starts an experiment, the *Envoy* starts the *Collaborator* to train the global model.

These components stay available to distribute several of experiments in the federation.

.. _openfl_ll_components_director:

Director
--------

The Director is a long-lived entity and is the central node of the federation. It accepts connections from: 

    - Frontend clients (data scientists using :ref:`interactive_api`)

>     - Envoys, if their Shar Descriptors are complient to the same data interface

Reviewer: Shard Descriptors

Author: done

The Envoy is a long-lived entity that runs on collaborator nodes connected to the Director. 

Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ 
in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, 
prepares the environment, and starts a Collaborator.

> Envoy is also responsible for sending heartbeat messages to the Director, this messages may include information 

Reviewer: The Envoy ... to the Director. This messages ...

> Envoy is also responsible for sending heartbeat messages to the Director. These messages may also include information 

Reviewer: ```suggestion
The envoy is also responsible for sending heartbeat messages to the Director. These messages may also include information 
```

> regarding the collaborator machine resources utilization. 

Reviewer: probably `regarding collaborator machine resource utilization.`?

> regarding the collaborator machine resources utilization. Refer to `device_monitor_plugin`_ for details.

Reviewer: ```suggestion
regarding collaborator machine resource utilization. Refer to `device_monitor_plugin`_ for details.
```
